custom-recipes/send-mails-from-contacts-dataset/recipe.py

```diff
@@ -118,25 +126,37 @@ def confirm_string_column(column_name, schema):
 if not is_subject_present:
     raise AttributeError("No value provided for the subject")
 
+is_sender_present = False
+
+if channel_has_sender:
+    is_sender_present = True
+else:
+    if use_sender_value:
+        is_sender_present = bool(sender_value)
+    else:
+        is_sender_present = bool(sender_column)
+if not is_sender_present:
+    raise AttributeError("No value provided for the sender")
+
```

Comment:
Explaining patch changes: added validation of sender being present as it was missing